feature_engine/discretisation/binariser.py

```diff
@@ -0,0 +1,231 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._check_init_parameters.check_variables import (
+    _check_variables_input_value,
+)
+from feature_engine._docstrings.fit_attributes import (
+    _binner_dict_docstring,
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.init_parameters.all_trasnformers import (
+    _variables_numerical_docstring,
+)
+from feature_engine._docstrings.init_parameters.discretisers import (
+    _precision_docstring,
+    _return_boundaries_docstring,
+    _return_object_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_discretiser_docstring,
+    _fit_transform_docstring,
+    _transform_discretiser_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+
+
+@Substitution(
+    return_object=_return_object_docstring,
+    return_boundaries=_return_boundaries_docstring,
+    precision=_precision_docstring,
+    binner_dict_=_binner_dict_docstring,
+    fit=_fit_discretiser_docstring,
+    transform=_transform_discretiser_docstring,
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class BinaryDiscretiser(BaseDiscretiser):
+    """
+    The BinaryDiscretiser() divides continuous numerical variables into two intervals,
+    where the value `threshold`, the point at which the interval is  divided, is
+    determined by the user.
+
+    The BinaryDiscretiser() works only with numerical variables.
+    A list of variables can be passed as argument. Alternatively, the discretiser
+    will automatically select all numerical variables.
+
+    The BinaryDiscretiser() first finds the boundaries for the intervals for
+    each variable. Then, it transforms the variables, that is, sorts the values into
+    the intervals.
+
+    Parameters
+    ----------
+    {variables}
+
+    threshold: int, float, default=None
+        Desired value at which to divide the interval.
+
+    {return_object}
+
+    {return_boundaries}
+
+    {precision}
```

Comment:
I also think we don't need precision, because we don't have interval boundaries. The boundaries are -inf to threshold and then threshold to inf. We can drop this one as well.